src/Rules/PhpDoc/IncompatiblePhpDocTypeRule.php

```diff
@@ -137,6 +140,20 @@ public function processNode(Node $node, Scope $scope): array
 						),
 					));
 
+					if ($phpDocParamType instanceof CallableType || $phpDocParamType instanceof ClosureType) {
```

Comment:
these lines cause phpstan to report an error when analysing itself but i wasn't sure of a safe alternative, given i've had to add `getTemplateTags` to these 2 types, and it's not safe to assume that `ParametersAcceptor` will have these methods

Reply:
Yes, I agree with this

Reply:
But you need to regenerate the baseline :)

Reply:
thought as much, just wanted to give you a chance to tell me if i was wrong

Reply:
Hey sorry, I was on my phone and thought we're in TemplateTypeHelper::toArgument (there's also `if ($type instanceof ParametersAcceptor) {`).

This has the same problem we've had there - this will only work with top-most type. We need to employ TypeTraverser again. (In TypeTraverser calls `instanceof CallableType` is allowed so baseline can be cleared again).

For example if you check what `$this->unresolvableTypeHelper->containsUnresolvableType($phpDocReturnType)` does, it looks for types recursively.

We should do a similar thing here :)

Reply:
ah yeah of course, great catch